modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/ExecutionContext.java

```diff
@@ -310,6 +311,10 @@ public void execute(RunnableX task, Consumer<Throwable> onError) {
             } catch (Throwable e) {
                 onError.accept(e);
 
+                if (e instanceof IgniteException) {
```

Comment:
perhaps, we should not re-throw exceptions here at all. WDYT?

Reply:
agree, thanks for the note